### PR TITLE
Feature ta 2739

### PR DESCRIPTION
**Pull request by averysmithproductions**
for jira ticket https://us-sba.atlassian.net/browse/TA-2739

- [x] Content API accepts query parameter for Keyword
- [x] Content API returns results with events containing those keywords
- [x] Exact Match search terms (search term should be found within the event's fields)
- [x] Search in Title, Description and Summary

### CODE REVIEW

File: src/service/event-search.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    sort: 'startdatetime asc',
+    queryParser: 'structured',
+    size: pageSize || defaultPageSize,
+    start: start || defaultStart
+  }
+  if (latitude && longitude) {
+    params = Object.assign({}, params, {
+      sort: 'distance asc',
+      return: '_all_fields,distance',
+      expr: `{"distance":"haversin(${latitude},${longitude},geolocation.latitude,geolocation.longitude)"}`
```
Comment (averysmithproductions) [inline src/service/event-search.js]: The event-search.js file is a fork of the office-search.js file.

Anything that references distance or geolocation stuff purposely does not work.

I decided to keep in pieces that reference geolocation, such as this line, so as to make future tickets that reference distance functionality easier to implement.

File: src/content.js [hunk 2]
```diff
@@ -18,8 +18,11 @@ const { getAuthors } = require('./service/authors.js')
 const { fetchBlogs, fetchBlog } = require('./service/blogs.js')
 const { fetchCourses, fetchCourse } = require('./service/courses.js')
 const { fetchDocuments } = require('./service/documents.js')
-const { fetchEvents, fetchEventById } = require('./service/events.js')
 const { fetchOffices } = require('./service/office-search.js')
+const d7Events = require('./service/events.js')
+const d8Events = require('./service/event-search.js')
+const events = config.eventsApi.getBackendSourceToggle() !== 'true' ? d7Events : d8Events
```
Comment (amyd11) [inline src/content.js]: interesting set up

File: src/service/office-search.js [hunk 3]
```diff
@@ -16,6 +16,7 @@ let { computeLocation } = require('./location.js')
 
 // for testing purposes
 async function runSearch (params) {
+  console.log('EE--', params)
```
Comment (amyd11) [inline src/service/office-search.js]: don't forget to remove this

File: src/service/event-search.js (hunk 1 above)
Comment (amyd11) [reply src/service/event-search.js]: I think this is a nice thought, but I do think we should comment out this section of code and add a comment above it alluding to what you just said.

File: src/service/event-search.js [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+  const result = await csd.search(params).promise()
+  return result
+}
+
+function buildQuery (query) {
+  const queryStatements = []
+  let queryString = `startdatetime: ['${moment.utc().format()}',}`
+
+  if (query) {
+    const fieldsToSearch = ['description', 'name', 'summary']
```
Comment (amyd11) [inline src/service/event-search.js]: cool, so we want to search description, name, and summary for keyword

File: src/service/event-search.js (hunk 1 above)
Comment (averysmithproductions) [reply src/service/event-search.js]: done

File: src/service/office-search.js (hunk 3 above)
Comment (averysmithproductions) [reply src/service/office-search.js]: done

File: src/service/event-search.js [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+    queryParser: 'structured',
+    size: pageSize || defaultPageSize,
+    start: start || defaultStart
+  }
+  /*
+  The event-search.js file is a fork of the office-search.js file.
+
+  Anything that references distance or geolocation stuff purposely does not work.
+
+  I decided to keep in pieces that reference geolocation, such as this line, so as to make future tickets that reference distance functionality easier to implement.
```
Comment (PatKelsh) [inline src/service/event-search.js]: Not a bad idea

File: src/service/event-search.test.js [hunk 6]
```diff
@@ -0,0 +1,93 @@
+/* eslint-env mocha */
+/* eslint-disable no-unused-expressions */
+let sinon = require('sinon')
+let chai = require('chai')
+chai.should()
+let eventSearch = require('./event-search.js')
+let mockCloudSearchResponseWithEvents = {
```
Comment (PatKelsh) [inline src/service/event-search.test.js]: We could put this in another file so we can expand out the test results without having to blow up the size of this file

**Pull request merged**